src/buildercore/cfngen.py

```diff
@@ -350,34 +372,46 @@ def generate_stack(pname, **more_context):
     """given a project name and any context overrides, generates a Cloudformation
     stack file, writes it to file and returns a pair of (context, stackfilename)"""
     context = build_context(pname, **more_context)
-    template = render_template(context)
+    cloudformation_template = render_template(context)
+    terraform_template = terraform.render(context)
     stackname = context['stackname']
-    out_fname = write_template(stackname, template)
+
     context_handler.write_context(stackname, context)
-    return context, out_fname
+    cloudformation_template_file = write_template(stackname, cloudformation_template)
+    terraform_template_file = write_terraform_template(stackname, terraform_template)
+    return context, cloudformation_template_file, terraform_template_file
 
 #
 # update existing template
 #
 
 
 # can't add ExtDNS: it changes dynamically when we start/stop instances and should not be touched after creation
-UPDATABLE_TITLE_PATTERNS = ['^CloudFront.*', '^ElasticLoadBalancer.*', '^EC2Instance.*', '.*Bucket$', '.*BucketPolicy', '^StackSecurityGroup$', '^ELBSecurityGroup$', '^CnameDNS.+$', '^AttachedDB$', '^AttachedDBSubnet$', '^ExtraStorage.+$', '^MountPoint.+$', '^IntDNS.*$', '^ElastiCache.*$']
+UPDATABLE_TITLE_PATTERNS = ['^CloudFront.*', '^ElasticLoadBalancer.*', '^EC2Instance.*', '.*Bucket$', '.*BucketPolicy', '^StackSecurityGroup$', '^ELBSecurityGroup$', '^CnameDNS.+$', 'FastlyDNS\\d+$', '^AttachedDB$', '^AttachedDBSubnet$', '^ExtraStorage.+$', '^MountPoint.+$', '^IntDNS.*$', '^ElastiCache.*$']
 
-REMOVABLE_TITLE_PATTERNS = ['^CnameDNS\\d+$', '^ExtDNS$', '^ExtraStorage.+$', '^MountPoint.+$', '^.+Queue$', '^EC2Instance.+$', '^IntDNS.*$', '^ElastiCache.*$', '^.+Topic$']
+REMOVABLE_TITLE_PATTERNS = ['^CnameDNS\\d+$', 'FastlyDNS\\d+$', '^ExtDNS$', '^ExtraStorage.+$', '^MountPoint.+$', '^.+Queue$', '^EC2Instance.+$', '^IntDNS.*$', '^ElastiCache.*$', '^.+Topic$']
```

Comment:
Just the usual whitelisting saying we can update the CNAME DNS entries of Fastly that are managed through CloudFormation.